Content/Items/Misc/SoilgunFiles/Weapons.Soilgun.cs

```diff
@@ -91,34 +91,34 @@ class SoilgunGlobalItem : GlobalItem
-                    case ItemID.EbonsandBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun it will fire out blocks of Ebonsand that apply stacks of Haunted to enemies"); break;
-                    case ItemID.PearlsandBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of Pearlsand that home in on enemies"); break;
-                    case ItemID.DirtBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of dirt"); break;
-                    case ItemID.SiltBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of silt, that spawn coins upon hitting enemies"); break;
-                    case ItemID.SlushBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of slush that cause hit enemies to have icicles impale them\nHitting and enemy with more than 10 icicles causes all icicles to shatter, causing large amounts of damage"); break;
-                    case ItemID.MudBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of mud that bounce off tiles and enemies"); break;
+                    TooltipLine tooltip = new TooltipLine(Mod, "SoilgunAmmoTooltip", "This item can be used as ammo for the Soilgun and Earthduster");
+                    tooltips.Add(tooltip);
+                    tooltip.OverrideColor = new Color(202, 148, 115);
+                    if (item.type == Mod.Find<ModItem>("VitricSandItem").Type)
+                    {
+                        TooltipLine infoTooltip = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun or Earthduster, it will fire out blocks of glassy sand, that cause crystals to grow out of enemies\nFor each crystal an enemy has, they take 2 damage per second, plus a base damage of 4, up to a maximum of 10 crystals\nIf an enemy has had 10 crystals on them for more than 4 seconds, all crystals become charged, exploding shorty after");
+                        tooltips.Add(infoTooltip);
+                        infoTooltip.OverrideColor = new Color(202, 148, 115);
+                        return;
+                    }
+                    switch (item.type)
+                    {
+                        case ItemID.SandBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun or Earthduster, it will fire out blocks of sand that split into many grains of sand upon death"); break;
+                        case ItemID.CrimsandBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun or Earthduster, it will fire out blocks of Crimsand that steal life from hit enemies"); break;
```

Comment:
This effect is really strong compared to the other ammos, This can be compared directly to vampire knives which are the other equivalent of lifesteal. vampire knives deal 7.5% lifesteal w/ 29 base damage, this deals 10% lifesteal w/ 25 base damage. vampire knives still edge out on dps/healing cause of the shotgun effect, but this has higher range making it sort of a vampire knife sidegrade. The other ammos are within the realm of reason for a late PHM, semi upgrade to minishark weapon, but this ammo type makes this weapon on par with a post-plantera weapon